src/audio/copier/copier.c

```diff
@@ -1025,18 +1025,21 @@ static int copier_comp_trigger(struct comp_dev *dev, int cmd)
 	if (ret == COMP_STATUS_STATE_ALREADY_SET)
 		return PPL_STATUS_PATH_STOP;
 
-	if (dev->ipc_config.type == SOF_COMP_HOST && !cd->ipc_gtw) {
-		ret = host_zephyr_trigger(cd->hd, dev, cmd);
-		if (ret < 0)
-			return ret;
-	}
-
-	for (i = 0; i < cd->endpoint_num; i++) {
-		if (dev->ipc_config.type != SOF_COMP_HOST || cd->ipc_gtw) {
-			ret = cd->endpoint[i]->drv->ops.trigger(cd->endpoint[i], cmd);
-			if (ret < 0)
-				break;
-		}
+	switch (dev->ipc_config.type) {
+	case SOF_COMP_HOST:
+		if (!cd->ipc_gtw)
+			ret = host_zephyr_trigger(cd->hd, dev, cmd);
+		else
+			/* handle gtw case */
+			ret = cd->endpoint[0]->drv->ops.trigger(cd->endpoint[0], cmd);
+		break;
+	case SOF_COMP_DAI:
+		ret = dai_zephyr_trigger(cd->dd, cd->endpoint[0], cmd);
+		break;
+	default:
+		comp_err(dev, "Unexpected copier device type to trigger! %d",
```

Comment:
ahh i forgot a module copier is not of type DAI or HOST. Please remove all errors as there's just nothing to do for that case.

Reply:
ok, remove error is fine, for moduler copier, what's the assigned type, or it does not need a type?